src/layouts/Workspace/FoldersAndPagesController.tsx

```diff
@@ -0,0 +1,89 @@
+import { VStack, GridItem, StackDivider } from "@chakra-ui/react"
+
+import { useGetFolders } from "hooks/directoryHooks"
+
+import { PageData } from "types/directory"
+
+import {
+  EmptyPageAndFolder,
+  EmptyFolder,
+  UngroupedPages,
+  Folders,
+  EmptyPage,
+} from "./WorkspacePagesAndFoldersComponents"
+
+export const FoldersAndPagesController = (props: {
+  siteName: string
+  url: string
+  pagesData: PageData[]
+}): JSX.Element => {
+  const { siteName, url, pagesData } = props
+  const { data: _dirsData } = useGetFolders({ siteName })
+  const dirsData = _dirsData || []
+  const isPagesEmpty =
+    pagesData.filter((page) => page.name !== "contact-us.md").length === 0
+  const isFoldersEmpty = !dirsData || dirsData.length === 0
+
+  if (isPagesEmpty && isFoldersEmpty) {
+    return <EmptyPageAndFolder url={url} />
+  }
+
+  if (isFoldersEmpty) {
+    return (
+      <GridItem
+        area="content"
+        as={VStack}
+        spacing="2rem"
+        bgColor="gray.50"
+        w="100%"
+        h="100%"
+        divider={<StackDivider borderColor="border.divider.alt" />}
+      >
```

Comment:
we seem to be redefining this for everything as a workaround for having the controller. this seems abit awkward and maybe we should have a context which the pages/folders both read from and then render accordingly

Reply:
I understand that you wanted to remove the `FoldersAndPagesController` altogether. I have added an [experimental branch here](https://github.com/isomerpages/isomercms-frontend/blob/temp/experiment-removing-workspace-controller/src/layouts/Workspace/Workspace.tsx) (to be removed) 
I originally preferred against it as nested inline conditions might be hard to read (if you are ok with this, I will just cherry-pick this commit over to this branch). Please do let me know if you have other way of refactoring this to remove the controller entirely!